### PR TITLE
feat: adding 09-localhost loopback client module

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

The following PR adds `v2` of the `09-localhost` light client module.

closes: #3034 

### Commit Message / Changelog Entry

```bash
feat: adding 09-localhost loopback client module [\#3034](https://github.com/cosmos/ibc-go/issues/3034). Please see the 09-localhost documentation [here](https://ibc.cosmos.network/main/ibc/light-clients/localhost/overview.html).
feat: AllowedClients on-chain param allows chains to pause usage of specific client types by removing the client type from the param.
imp: Update all channel events to use `connection_id` attribute. The `packet_connection` attribute has been deprecated.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/relayer/relayer.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ const (
 	Rly    = "rly"
 	Hermes = "hermes"
 
-	cosmosRelayerRepository = "ghcr.io/cosmos/relayer"
-	cosmosRelayerUser       = "100:1000" // docker run -it --rm --entrypoint echo ghcr.io/cosmos/relayer "$(id -u):$(id -g)"
+	cosmosRelayerRepository = "damiannolan/rly" //"ghcr.io/cosmos/relayer"
```
Comment (damiannolan) [inline e2e/relayer/relayer.go:18]: https://github.com/cosmos/relayer/issues/1102

**Review by AdityaSripal — APPROVED** [review]

**Review by chatton — COMMENTED** [review]
Maybe we can add a link to where the docs will be in the commit message?

And link to the localhost issue with a breakdown of all sub issues

File: docs/migrations/v7-to-v7_1.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  - [Relayers](#relayers)
+  - [IBC Light Clients](#ibc-light-clients)
+
+**Note:** ibc-go supports golang semantic versioning and therefore all imports must be updated on major version releases.
+
+## Chains
+
+In the previous release of ibc-go, the localhost `v1` light client module was deprecated and removed. The ibc-go `v7.1.0` release introduces `v2` of the 09-localhost light client module.
+
+<!-- TODO: Update the links to use release version instead of feat branch -->
```
Comment (chatton) [inline docs/migrations/v7-to-v7_1.md:19]: is there an issue for this? This will be a dead link as soon as the branch is deleted.

**Review by colin-axner — APPROVED** [review]
:)

**Review by chatton — APPROVED** [review]
LGTM 🥳 🥳 🎈

File: docs/migrations/v7-to-v7_1.md (hunk 2 above)
Comment (chatton) [reply docs/migrations/v7-to-v7_1.md:19]: will create one for this so we don't lose it if it doesn't exist.

File: .github/workflows/e2e-upgrade.yaml [hunk 3]
```diff
@@ -41,3 +41,15 @@ jobs:
...
       test-entry-point:  "TestUpgradeTestSuite"
       test: "TestV6ToV7ChainUpgrade"
+
+  upgrade-v7_1:
+    uses: cosmos/ibc-go/.github/workflows/e2e-test-workflow-call.yml@main
+    with:
+      chain-image: ghcr.io/cosmos/ibc-go-simd
+      chain-binary: simd
+      chain-a-tag: pr-3136 # TODO: needs v7.0.0 (with simapp fixes) when cut
+      chain-b-tag: pr-3136 # TODO: needs v7.0.0 (with simapp fixes) when cut
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply .github/workflows/e2e-upgrade.yaml:51]: #3261

**Pull request merged**